### ISSUE TITLE
BigTable: row.commit throwing _Rendezvous exception instead of ServiceUnavailable

### ISSUE THREAD

**Issue by zakons** [GoogleCloudPlatform/google-cloud-python#4656]
We are using retry.py and creating a retry exception predicate as follows:

predicate=retry.if_exception_type(exceptions.ServiceUnavailable)

However, since we are only getting _Rendezvous exceptions with StatusCode.UNAVAILABLE (see below stack trace), we had to change this to:

predicate=retry.if_exception_type(exceptions.GrpcRendezvous)

I see a different issue in cloud datastore for mapping gRPC codes to service level exceptions: 

https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2590

Will this also be done for cloud bigtable?

Stack Trace:

  File "single_key_read_write.py", line 135, in write
    row.commit()
  File "/home/jenkins/.local/lib/python3.5/site-packages/google/cloud/bigtable/row.py", line 445, in commit
    retry_(retry_commit)()
  File "/home/jenkins/.local/lib/python3.5/site-packages/google/api_core/retry.py", line 261, in retry_wrapped_func
    on_error=on_error,
  File "/home/jenkins/.local/lib/python3.5/site-packages/google/api_core/retry.py", line 178, in retry_target
    return target()
  File "/home/jenkins/.local/lib/python3.5/site-packages/google/cloud/bigtable/row.py", line 256, in __call__
    client._data_stub.MutateRow(self.request_pb)
  File "/home/jenkins/.local/lib/python3.5/site-packages/grpc/_channel.py", line 492, in __call__
    return _end_unary_response_blocking(state, call, False, deadline)
  File "/home/jenkins/.local/lib/python3.5/site-packages/grpc/_channel.py", line 440, in _end_unary_response_blocking
    raise _Rendezvous(state, None, None, deadline)
grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with (StatusCode.UNAVAILABLE, Transport closed)>

# Thank you for reporting an issue to google-cloud-python!

If you are reporting an issue or requesting a feature, please search the
existing open and closed issues to see if there is already work being done.

- https://github.com/GoogleCloudPlatform/google-cloud-python/issues
- http://stackoverflow.com/questions/tagged/google-cloud-python

If you can provide us with as much of the following information as possible it
will help us identify the cause of your issue more quickly.

1. Specify the API at the beginning of the title (for example, "BigQuery: ...")
   General, Core, and Other are also allowed as types
2. OS type and version
3. Python version and virtual environment information `python --version`
4. google-cloud-python version `pip show google-cloud`, `pip show google-<service>` or `pip freeze`
5. Stacktrace if available
6. Steps to reproduce
7. Code example

Using GitHub flavored markdown can help make your request clearer.
See: https://guides.github.com/features/mastering-markdown/

**Comment by dhermes** [comment]
@zakons In order for the "new" exceptions to be in place, the Bigtable gRPC / GAPIC surface will need to be regenerated.

@lukesneeringer Can we make this happen?

**Comment by sduskis** [comment]
@dhermes: IIUC, the Bigtable client doesn't actually use GAPIC.

FWIW, I'm all for regenerating the protobuf and gRPC artifacts.  I'd also be happy if we can migrate to GAPIC.

**Comment by dhermes** [comment]
> IIUC, the Bigtable client doesn't actually use GAPIC.

That's correct. Bigtable was the **very first** package to use `protoc` with the `grpc` extension and we still have a `Makefile` to build the `google.cloud.bigtable._generated` package.

This **very much** should be retired in favor of the "modern" autogen infrastructure.

**Comment by zakons** [comment]
Thanks.  Just to clarify, after the regeneration we would get a mapping of the Rendezvous exception to a ServiceUnavailable exception?  Curious to know how this code generation works so that the code in api_core/google/api_core/exceptions.py for ServiceUnavailable is used.

**Comment by jonparrott** [comment]
> Just to clarify, after the regeneration we would get a mapping of the Rendezvous exception to a ServiceUnavailable exception?

Yes

> Curious to know how this code generation works so that the code in api_core/google/api_core/exceptions.py for ServiceUnavailable is used.

The generator generates wrappers for the underlying grpc client (*stub* in grpc terminology). One of the wrappers applied to all RPC methods is [`google.api_core.grpc_helpers.wrap_errors`](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/e58fecd99f2f19ed5fb1026cdfd178ae3f6d8da2/api_core/google/api_core/grpc_helpers.py#L90) which handles re-mapping `grpc.RpcErrors` to appropriate exception classes. `wrap_errors` is applied to RPC methods by [`google.api_core.gapic_v1.method.wrap_method`](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/api_core/google/api_core/gapic_v1/method.py#L219) which is called by the generated code ([example](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/pubsub/google/cloud/pubsub_v1/gapic/publisher_client.py#L130)).

**Comment by tseaver** [comment]
@jonparrott I'm not sure if it is you or @lukesneeringer who needs to do the non-re-generation of the GAPIC source from the protos.
